src/shared/Microsoft.Git.CredentialManager/PlatformUtils.cs

```diff
@@ -20,6 +20,26 @@ public static PlatformInformation GetPlatformInformation()
             return new PlatformInformation(osType, cpuArch, clrVersion);
         }
 
+        public static bool IsWindows10()
+        {
+            if (!IsWindows())
+            {
+                return false;
+            }
+
+            // Implementation of version checking was taken from:
+            // https://github.com/dotnet/runtime/blob/6578f257e3be2e2144a65769706e981961f0130c/src/libraries/System.Private.CoreLib/src/System/Environment.Windows.cs#L110-L122
```

Comment:
Very cool,. we use a nastier approach based on Registry for netframework.

Reply:
Yeah I thought there has to be a better solution given the .NET folks invented all those `OperatingSystem.IsWindowsVersionOrGreater` etc APIs, and they correctly identify Windows 10 without manifests.